zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -203,19 +203,24 @@ mod tests {
         let re = Regex::new(r"\s+").unwrap();
 
         let expected_tables = vec![
-            "CREATE TABLE \"accounts\" (
-                account INTEGER PRIMARY KEY,
-                ufvk TEXT NOT NULL,
+            r#"CREATE TABLE "accounts" (
+                id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+                account_type INTEGER NOT NULL DEFAULT 0,
+                hd_seed_fingerprint BLOB,
+                hd_account_index INTEGER,
+                uvk TEXT NOT NULL,
                 birthday_height INTEGER NOT NULL,
-                recover_until_height INTEGER )",
-            "CREATE TABLE addresses (
+                recover_until_height INTEGER,
+                CHECK ( (account_type = 0 AND hd_seed_fingerprint IS NOT NULL AND hd_account_index IS NOT NULL) OR (account_type = 1 AND hd_seed_fingerprint IS NULL AND hd_account_index IS NULL) )
+            )"#,
+            r#"CREATE TABLE "addresses" (
```

Comment:
Since we're having to rewrite these tables in order to update the foreign key constraints, I would prefer we at the same time adopt the convention of using `account_id` for the name of the key column that references `accounts(id)`. I realize that this requires a bit more of an invasive change to make sure we update queries, but I think it's worth it to use a consistent convention.

Reply:
That sounds good to me. IIRC one table had *two* FKs into the accounts table (payments to/from account). Do you want these columns suffixed with `_account_id`?

Reply:
The only schema changes that go into the CHANGELOG are those that affect the views; the views are currently considered part of the public API, whereas the tables themselves are not.